packages/material-ui-styles/src/ThemeProvider/nested.js

```diff
@@ -1,3 +1,3 @@
-const hasSymbol = typeof Symbol === 'function';
+const hasSymbol = typeof Symbol === 'function' && Symbol.for;
 
 export default (hasSymbol ? Symbol.for('mui.nested') : '__THEME_NESTED__');
```

Comment:
I'm just hijacking this to fill in my own gaps: What's the point of a global Symbol.for registry? If we fallback to strings in bad polyfills anyway why not just use them? Is this some micro optimization I'm missing?

Reply:
@eps1lon No specific reason, I have "almost" blindly copied the react approach. I think that the point is to make it internal, it's not something we want people to hack around. Symbol.for should be supported in 99% of the developers' environment even if this number is lower in end users.

Reply:
Yeah no this is more about what `Symbol.for` is useful since it's anything but internal. The mdn docs even recommend prefixing it to avoid name collisions. What does `Symbol.for('mui.nested')` do that `'mui.nested'` can't?

Reply:
> What does Symbol.for('mui.nested') do that 'mui.nested' can't?

I would say, let's compare 
```jsx
var theme1 = {
  [Symbol.for('mui.nested')]: true,
}

var theme2 = {
  'mui.nested': true,
}

console.log(theme1, theme2)
```

variant 1 doesn't make me want to hack with the value.

Reply:
The second.

Reply:
On my end, I would avoid both 😆. As for the less interested in hacking, I have no strong preference either way.

Reply:
@briandelancey what's your perspective on the issue?

Reply:
I was not proposing to deviate. When in doubt assume the author added it with intent. Just thought anybody know what technical benefits `Symbol.for` has.

Reply:
@eps1lon ok, sounds good to me. Is it ready to ship?

Reply:
Thanks for raising this point.

Reply:
@oliviertassinari I'm ok with https://stackoverflow.com/questions/51488519/what-advantage-of-using-symbol-forstring-instead-the-plain-string

Not being enumerate by the common `Object.keys` seams reasonable. And since the theme is often passed around (potentially stringified for hydrated UIs) having it not appear for `JSON.stringify` serialization is also nice. So it's better at emulating internal properties.